cmd/mistryd/worker.go

```diff
@@ -191,12 +191,17 @@ func (s *Server) Work(ctx context.Context, j *Job) (buildInfo *types.BuildInfo,
 
 	var outErr strings.Builder
 	buildInfo.ExitCode, err = j.StartContainer(ctx, s.cfg, client, out, &outErr)
-
 	if err != nil {
 		err = workErr("could not start docker container", err)
 		return
 	}
 
+	err = client.Close()
+	if err != nil {
+		err = workErr("could not close docker client", err)
+		return
+	}
+
```

Comment:
Maybe we need to defer this? As it is now, if `BuildImage` or `StartContainer` errors, it will never run.

Reply:
True, I handled it with a defer function.